bloom.go

```diff
@@ -121,14 +127,35 @@ func (f *Filter[T, H]) Test(data T) bool {
 	return true
 }
 
+func (f *Filter[T, H]) checkCompatibility(other *Filter[T, H]) error {
+	if reflect.ValueOf(f.function).Pointer() != reflect.ValueOf(other.function).Pointer() {
```

Comment:
My understanding is that reflection comes at a perf cost. For something like bloom filters where perf matters, would it be better to skip this reflect usage? 🤔